src/main/scala/loamstream/googlecloud/CloudSdkDataProcWrapper.scala

```diff
@@ -0,0 +1,92 @@
+package loamstream.googlecloud
+
+import loamstream.util.Loggable
+import loamstream.util.Paths.normalize
+import scala.util.Try
+import scala.util.Success
+import loamstream.util.Tries
+import loamstream.util.ExitCodes
+import scala.concurrent.duration.Duration
+import loamstream.util.BashScript
+import java.nio.file.Paths
+
+/**
+ * @author clint
+ * Nov 28, 2016
+ */
+final class CloudSdkDataProcWrapper private[googlecloud] (
+  config: GoogleCloudConfig,
+  runCommand: Seq[String] => Int = CloudSdkDataProcWrapper.runCommand) extends 
+      DataProcClient.CanStop with DataProcClient.CanTellIfRunning with Loggable {
+
+  import CloudSdkDataProcWrapper.{
+    deleteClusterTokens,
+    isClusterRunningTokens
+  }
+
+  override def stopCluster(): Unit = {
+    debug(s"Deleting cluster '${config.clusterId}'")
+
+    val exitCode = runCommand(deleteClusterTokens(config))
+
+    ExitCodes.throwIfFailure(exitCode)
+  }
+
+  override def isClusterRunning: Boolean = {
+    val exitCode = runCommand(isClusterRunningTokens(config))
+
+    val result = ExitCodes.isSuccess(exitCode)
+
+    debug(s"Cluster '${config.clusterId}' running? $result")
+
+    result
+  }
+}
+
+object CloudSdkDataProcWrapper extends Loggable {
+  def fromConfig(config: GoogleCloudConfig): Try[CloudSdkDataProcWrapper] = {
+    val gcloudBinary = config.gcloudBinary.toFile
+
+    if (gcloudBinary.exists && gcloudBinary.canExecute) {
+      Success(new CloudSdkDataProcWrapper(config))
+    } else {
+      Tries.failure(s"gcloud executable not found at ${config.gcloudBinary} or not executable")
```

Comment:
How is `Tries.failure` different from `Failure`? Confused a bit since `Try` is in the standard library and `Tries` is not?

Reply:
Side note: when I see `Tries` I think of string trees: https://en.wikipedia.org/wiki/Trie 😄

Reply:
`Tries.failure` is just a helper method that makes a `Failure[A]` wrapping a particular exception with a particular error message.  This lets you write 
```scala
Tries.failure("oof!")
```
instead of the more verbose
```scala
Failure(new Exception("oof!"))
```
The name `Tries` comes from the convention of putting "static" helper methods related to `<typename>` in a container called `<plural-form-of-typename>`, say `Foo`-related methods in `Foos`.  

I _also_ think of the data structure, but `Trys` looked funny, and something like, say, `Attempts` broke convention by being a different word entirely.